File: src/main/kotlin/com/full/stack/assignment/f1/model/Driver.kt
package com.full.stack.assignment.f1.model

import com.fasterxml.jackson.annotation.JsonProperty

data class Driver(
    @JsonProperty("driverId")
    val id: String,
    @JsonProperty("code")
    val code: String?,
    @JsonProperty("givenName")
    val firstName: String,
    @JsonProperty("familyName")
    val lastName: String,
    @JsonProperty("dateOfBirth")
    val dateOfBirth: String,
    @JsonProperty("nationality")
    val nationality: String
)

File: frontend/lib/domain/models/driver/driver.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'driver.freezed.dart';

@freezed
abstract class Driver with _$Driver {
  const Driver._();

  const factory Driver({
    required String driverId,
    required String? code,
    required String givenName,
    required String familyName,
    required String dateOfBirth,
    required String nationality,
  }) = _Driver;

  String get fullName => '$givenName $familyName';
}
